conda_build/build.py

```diff
@@ -302,6 +303,44 @@ def sanitize_channel(channel):
     return re.sub('\/t\/[a-zA-Z0-9\-]*\/', '/t/<TOKEN>/', channel)
 
 
+def write_info_files_file(m, files, config):
+    entry_point_scripts = m.get_value('build/entry_points')
+    entry_point_script_names = get_entry_point_script_names(entry_point_scripts)
+
+    mode_dict = {'mode': 'w', 'encoding': 'utf-8'} if PY3 else {'mode': 'wb'}
+    with open(join(config.info_dir, 'files'), **mode_dict) as fo:
+        if m.get_value('build/noarch_python'):
+            fo.write('\n')
+        elif is_noarch_python(m):
+            for f in files:
+                if f.find("site-packages") > 0:
+                    fo.write(f[f.find("site-packages"):] + '\n')
+                elif f.startswith("bin") and (f not in entry_point_script_names):
+                    fo.write(f.replace("bin", "python-scripts") + '\n')
+                elif f.startswith("Scripts") and (f not in entry_point_script_names):
+                    fo.write(f.replace("Scripts", "python-scripts") + '\n')
+        else:
+            for f in files:
+                fo.write(f + '\n')
+
+
+def write_package_metadata_json(m, config):
+    extra = dict()
+
+    noarch_type = m.get_value('build/noarch')
+    if noarch_type:
+        noarch_dict = dict(type=text_type(noarch_type))
+        if text_type(noarch_type).lower() == "python":
+            entry_points = m.get_value('build/entry_points')
+            if entry_points:
+                noarch_dict['entry_points'] = entry_points
+        extra['noarch'] = noarch_dict
```

Comment:
Should this also include a `version` and `preferred_envs` keys as per https://github.com/conda/conda-build/pull/1539#issuecomment-266322808 ?